components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -17,6 +17,7 @@ open class FullScreenFeature(
     sessionManager: SessionManager,
     private val sessionUseCases: SessionUseCases,
     private val sessionId: String? = null,
+    private val viewportFitChange: (String) -> Unit = {},
     private val fullScreenChanged: (Boolean) -> Unit
```

Comment:
```suggestion
    private val fullScreenViewPortChanged: (Boolean, String) -> Unit
```

Should we have a single callback from `viewportFitChange` events along with `fullScreenChanged`? I'm not sure if consumers will find it easy to handle those separately since they seem coupled together (e.g. when fullscreen changes, you would want to know the viewportFit, and vice versa).

The internal implementation would call the session value for the other (see below).

Reply:
The GeckoView API is separate though, if we coupled them then the consumers will get fullScreenChanged called twice?

Reply:
That's a good point.

When I think about this code from a consumer's side, it seems like we're giving them multiple state changes to be handled separately, but they would really need them all together either way to make correct decisions.

```kotlin
FullscreenFeature(// ...
  ::handleViewPortChanged,
  ::handleFullscreenChanged)

fun handleViewPortChanged(viewPort: String) {
  if (sessionManager.findSession().fullscreen) {
    if (viewPort == "contain") {
      // do fullscreen with viewport change
    }
  }
}

fun handleFullscreenChanged(fullscreen: Boolean) {
  if (fullscreen) {
    if (sessionManager.findSession().viewPort == "contain") {
      // do fullscreen with viewport change
    }
  } else {
    if (sessionManager.findSession().viewPort == "none") {
      // do non-fullscreen with viewport
    }
  }
}
```

The unknown in my knowledge is how apps want to handle the viewport value with and without fullscreen, so maybe the above code is the outcome regardless?

Reply:
I see.  This is my mistake in my explanation.  The consumers doesn't have to need them together to make the correct decision.  These two state changes can be handled separately by the consumers without knowing the other state.